numba/typing/cffi_utils.py

```diff
@@ -24,13 +26,22 @@ def is_cffi_func(obj):
     try:
         return ffi.typeof(obj).kind == 'function'
     except TypeError:
-        return False
+        try:
+            return obj in _ool_func_types
+        except:
+            return False
 
 def get_pointer(cffi_func):
     """
     Get a pointer to the underlying function for a CFFI function as an
     integer.
     """
+    if cffi_func in _ool_func_types:
+        # We have no record of the function's address in an OOL module, so we
+        # resort to looking up the function by name in the interpreter's
+        # symbol table using ctypes
+        cfunc = getattr(ctypes.pythonapi, cffi_func.__name__)
```

Comment:
That's a great suggestion, many thanks! Will update the PR and add your example as a test.